aten/src/ATen/native/cuda/TensorTransformations.cu

```diff
@@ -188,9 +189,11 @@ Tensor roll_cuda(const Tensor& self, IntArrayRef shifts, IntArrayRef dims) {
 
   auto total_dims = in_tensor.dim();
 
-  AT_DISPATCH_ALL_TYPES_AND2(at::ScalarType::Half, at::ScalarType::Bool, in_tensor.scalar_type(), "roll_cuda", [&] {
+  AT_DISPATCH_ALL_TYPES_AND_C10_COMPLEX_AND3(at::ScalarType::Half, at::ScalarType::Bool, at::ScalarType::BFloat16,
+    in_tensor.scalar_type(), "roll_cuda", [&] {
+    using value_t = typename ztype<scalar_t>::value_t;
```

Comment:
Why do we need a `value_t` here? CPU's `ztype<scalar_t>::value_t` is a noop for `c10::complex`

Reply:
yeah forgot to remove it after replacing `AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3` with `AT_DISPATCH_ALL_TYPES_AND_C10_COMPLEX_AND3`